File: Wander/Assets/Networking/Response/ResponseMovement.cs
using UnityEngine;
using System;

public class ResponseMoveEventArgs : ExtendedEventArgs {

    public string username { get; set; }
    public string locationX { get; set; }
	public string locationY { get; set; }
	public ResponseMoveEventArgs() {
		event_id = Constants.SMSG_MOVE;
	}

}

public class ResponseMovement : NetworkResponse {

    private string username;
    private string locationX;
	private string locationY;
    

	public ResponseMovement() {
	}

	public override void parse() {
        username = DataReader.ReadString(dataStream);
        locationX = DataReader.ReadString(dataStream);
		locationY = DataReader.ReadString(dataStream);
	}

	public override ExtendedEventArgs process() {
		ResponseMoveEventArgs args = null;
		args = new ResponseMoveEventArgs();
		args.username = username;
		args.locationX = locationX;
		args.locationY = locationY;
		return args;
	}

}

File: Wander/WoB_Server_Basic/src/networking/response/ResponseMovement.java
package networking.response;

// Other Imports
import metadata.Constants;
import utility.GamePacket;

/**
 * The ResponseLogin class contains information about the authentication
 * process.
 */
public class ResponseMovement extends GameResponse {

    private String locationX;
    private String locationY;
    private String username;

    public ResponseMovement() {
        responseCode = Constants.SMSG_MOVE;
    }

    @Override
    public byte[] constructResponseInBytes() {
        GamePacket packet = new GamePacket(responseCode);
        packet.addString(username);
        packet.addString(locationX);
        packet.addString(locationY);
        return packet.getBytes();
    }

    public void setLocationX (String locationX) {
        this.locationX = locationX;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public void setLocationY (String locationY) {
        this.locationY = locationY;
    }
}